chem/module_optical_averaging.F

```diff
@@ -3692,47 +3695,51 @@ subroutine optical_prep_gocart(nbin_o, chem, alt,relhum,          &
         end do
-!  real, save :: seasfrc_goc8bin(4,nbin_o)   ! GOCART seasalt size distibution - mass fracs in MOSAIC 8-bins
-!  real, save :: dustfrc_goc8bin(ndust,nbin_o)   ! GOCART dust size distibution - mass fracs in MOSAIC 8-bins
-!  USE module_data_gocart_seas
-! real*8, DIMENSION (4), PARAMETER :: ra(4)=(/1.d-1,5.d-1,1.5d0,5.0d0/)
-! real*8, DIMENSION (4), PARAMETER :: rb(4)=(/5.d-1,1.5d0,5.d0,1.d1/)
-! real*8, DIMENSION (4), PARAMETER :: den_seas(4)=(/2.2d3,2.2d3,2.2d3,2.2d3/)
-! real*8, DIMENSION (4), PARAMETER :: reff_seas(4)=(/0.30D-6,1.00D-6,3.25D-6,7.50D-6/)
-!  USE module_data_gocart_dust, only:  ndust, reff_dust, den_dust
-!  real*8, DIMENSION (5), PARAMETER :: den_dust(5)=(/2500.,2650.,2650.,2650.,2650./)
-!  real*8, DIMENSION (5), PARAMETER :: reff_dust(5)=(/0.73D-6,1.4D-6,2.4D-6,4.5D-6,8.0D-6/)
 ! Seasalt bin mass fractions
-        seasfrc_goc8bin=0.
+        seasfrc_goc9bin=0.
 !      WRITE(*,*)'Seasalt mass fractions'
 !      WRITE(*,*)'         ','       ',(dlo_sectm(n),n=1,nbin_o)
 !      WRITE(*,*)'         ','       ',(dhi_sectm(n),n=1,nbin_o)
        do m =1, 4  ! loop over seasalt size bins
        dlogoc = ra(m)*2.E-6  ! low diameter limit (m)
        dhigoc = rb(m)*2.E-6  ! hi diameter limit (m)
         do n = 1, nbin_o
-        seasfrc_goc8bin(m,n)=max(DBLE(0.),min(DBLE(log(dhi_sectm(n))),log(dhigoc))- &
+        seasfrc_goc9bin(m,n)=max(DBLE(0.),min(DBLE(log(dhi_sectm(n))),log(dhigoc))- &
                              max(log(dlogoc),DBLE(log(dlo_sectm(n)))) )/(log(dhigoc)-log(dlogoc))
 
        end do
-!      WRITE(*,*)m,dlogoc,dhigoc,(seasfrc_goc8bin(m,n),n=1,nbin_o)
+!      WRITE(*,*)m,dlogoc,dhigoc,(seasfrc_goc9bin(m,n),n=1,nbin_o)
        end do
 ! Dust bin mass fractions
 !      WRITE(*,*)'Dust mass fractions'
 !      WRITE(*,*)'         ','       ',(dlo_sectm(n),n=1,nbin_o)
 !      WRITE(*,*)'         ','       ',(dhi_sectm(n),n=1,nbin_o)
-        dustfrc_goc8bin=0.
+        dustfrc_goc9bin=0.
        do m =1, ndust  ! loop over dust size bins
         dlogoc = ra_dust(m)*2.E-6  ! low diameter limit (m)
         dhigoc = rb_dust(m)*2.E-6  ! hi diameter limit (m)
         do n = 1, nbin_o
-        dustfrc_goc8bin(m,n)=max(DBLE(0.),min(DBLE(log(dhi_sectm(n))),log(dhigoc))- &
+        dustfrc_goc9bin(m,n)=max(DBLE(0.),min(DBLE(log(dhi_sectm(n))),log(dhigoc))- &
                              max(log(dlogoc),DBLE(log(dlo_sectm(n)))) )/(log(dhigoc)-log(dlogoc))
 
        end do
-!      WRITE(*,*)m,dlogoc,dhigoc,(dustfrc_goc8bin(m,n),n=1,nbin_o)
+!      WRITE(*,*)m,dlogoc,dhigoc,(dustfrc_goc9bin(m,n),n=1,nbin_o)
        end do
         kcall=kcall+1
+
+       !Diagnostic. A. Ukhov 09/17/24
+       !-----
+        WRITE(*,*)nbin_o
+        WRITE(*,*)'Dust redistribution:'
+        do m =1, ndust
+            WRITE(*,*)m,dustfrc_goc9bin(m,:)
+        end do
+
+        WRITE(*,*)'Sea salt redistribution:'
+        do m =1, 4
+            WRITE(*,*)m,seasfrc_goc9bin(m,:)
+        end do
+        !-----
```

Comment:
Unless this is controlled by a switch, these write statements should be removed or commented out for general use of the code.